packages/api-request-builder/src/query-id.js

```diff
@@ -62,3 +68,23 @@ export function byCustomerId(custId: string): Object {
   this.params.customerId = custId
   return this
 }
+
+/**
+ * Set the given `id` to the `cartId` internal state of the service instance.
+ * For querying shipping methods by cart id
+ *
+ * @param  {string} id - A resource `UUID`
+ * @throws If `id` is missing.
+ * @return {Object} The instance of the service, can be chained.
+ */
+export function byCartId(cartId: string): Object {
+  if (!cartId) throw new Error('Required argument for `byCartId` is missing')
+  if (this.params.id)
+    throw new Error(
+      'An ID for this resource has already been set. ' +
+        'You cannot use both `byId` and `byCartId`.'
+    )
```

Comment:
What happens when you set `byCartId` and `byCustomerId`? Are they mutually exclusive?

Reply:
I cannot figure out a case where a combination of a cart id and a customer id filter is valid to be honest. Not sure if we should restrict the case though.

Reply:
Ideally, if a customer has multiple carts (which is possible, at least on the API level), having both query parameters returns the cart that belongs to the customer. This same cart will be returned if queried with only the `cartId`, so I think we should not restrict it as it is a valid query parameter.

Reply:
Great, thanks for clarifying!